python/ray/data/dataset.py

```diff
@@ -2456,6 +2458,19 @@ def iter_batches(
                 to select ``numpy.ndarray`` for tensor datasets and
                 ``Dict[str, numpy.ndarray]`` for tabular datasets. Default is "native".
             drop_last: Whether to drop the last batch if it's incomplete.
+            local_shuffle_buffer_size: If non-None, the data will be randomly shuffled
+                using a local in-memory shuffle buffer, and this value will serve as the
+                minimum number of rows that must be in the local in-memory shuffle
+                buffer in order to yield a batch. This is a light-weight alternative to
+                the global `.random_shuffle()` operation; this shuffle will be less
+                random but will be faster and less resource-intensive. This buffer size
+                must be greater than or equal to ``batch_size``, and therefore
+                ``batch_size`` must also be specified when using local shuffling.
```

Comment:
OOC is this a requirement for the implementation to work, or imposed because having a buffer size smaller than the batch size results in close to 0 randomness?

Reply:
We discussed this a bit here: https://github.com/ray-project/ray/pull/26094#discussion_r906535370

It's required for the current implementation to work, but we could have a much simpler shuffling algorithm for the unbatched case. To keep the PR small and given that we're not aware of any use cases for unbatched local shuffling, we've been treating that as a P1 for a follow-up PR.